src/DependencyInjection/CompilerPass/AutoRegisterMessageHandlerCompilerPass.php

```diff
@@ -36,6 +36,18 @@ public function process(ContainerBuilder $container): void
             $aliasName = sprintf('handler_%s', $serviceId);
             $container->setAlias($aliasName, $serviceId)->setPublic(true);
 
+            $handlerClass = $container->getDefinition($serviceId)->getClass();
+            $messageClass = substr($handlerClass, 0, -7);
+
+            if (class_exists($messageClass)) {
+                $resolverDefinition->addMethodCall(
+                    'addHandler',
+                    [$messageClass, $aliasName]
+                );
+
+                continue;
+            }
+
             foreach ($tags as $tag) {
```

Comment:
Do we agree that a handler can only handle one message ?
If so then we should not allow multiple tags and we should throw an exception if there is more than one.

Then we should only try to resolve the message class with the FQCN if the tag doesn't contains it.
The auto discovery mechanism should be consider a fallback so we can override it in case of need.

Reply:
It depends on the handler type: Command and Query buses usually have 1 handler, but Event buses may have multiple handlers.

Reply:
Ici, il ne s'agit pas d'un event bus, et ce n'est pas non plus un système de saga. nous allons donc rester simple avec 1 handler par query/command.

Reply:
fixed